tests/testthat/test-volsurface.R

```diff
@@ -0,0 +1,19 @@
+context("VolSurface")
+
+testthat::test_that(desc = "vol surface creation works", {
+  testthat::expect_is(build_vol_surface(), "VolSurface")
+  }
+)
+
+testthat::test_that("time variance interpolation works", {
+  testthat::expect_true(is.LinearTimeVarInterpolation(LinearTimeVarInterpolation()))
+})
+
+testthat::test_that("the interpolation is working as expected", {
+  vs <- build_vol_surface()
+  testthat::expect_equal(interpolate_vol(vs, as.Date("2023-10-10"), 96)
+           , 0.26853916752886564)
+  testthat::expect_equal(interpolate_vol(vs, as.Date("2020-02-29"), 150),
+           0.19909034016558932)
+})
+
```

Comment:
This needs to be slightly more comprehensive. Take your queue from the `test-interpolate.R` file

Reply:
If I didn't misunderstood the comment, I have added 2 more interpolation points.

Reply:
yes, but can you also provide some interpolation points that include the region before the first strike and after the last strike / maturity date?

Reply:
Added a point with strike = 3, on the left of the interpolation support; and a point with strike = 300 on the right of the support with maturity = "2025-06-10" beyond the maturities of the surface data.